comit/src/network/protocols/orderbook.rs

```diff
@@ -338,6 +344,18 @@ impl Order {
     }
 }
 
+impl NewOrder {
+    pub fn assert_valid_ledger_pair(&self) -> anyhow::Result<()> {
+        let a = self.bitcoin_ledger;
+        let b = self.ethereum_ledger;
+
+        if ledger::is_valid_ledger_pair(a, b) {
+            return Ok(());
+        }
+        Err(anyhow::anyhow!("invalid ledger pair {}/{}", a, b))
```

Comment:
If you are only allowing "valid" ledger pairs, why don't you just have a single field of type `Kind` in the `Order`?
Then you can't even run into this error.

Reply:
I thought about doing that. The current solution allows folks to stipulate their preferred Ethereum testnet network.

Reply:
Fair point although I wonder if that flexibility doesn't harm us in the long run :)
Happy to move forward with this and see where we end up.